lib/friends/location.rb

```diff
@@ -23,21 +25,65 @@ def self.deserialization_expectation
     end
 
     # @param name [String] the name of the location
-    def initialize(name:)
+    def initialize(name:, nickname_str: nil)
       @name = name
+      @nicknames = nickname_str&.split(" #{NICKNAME_REFIX}") || []
     end
 
     attr_accessor :name
+    attr_reader :nicknames
 
     # @return [String] the file serialization text for the location
     def serialize
-      "#{SERIALIZATION_PREFIX}#{@name}"
+      Paint.unpaint("#{SERIALIZATION_PREFIX}#{self}")
     end
 
-    # @return [Regexp] the regex used to match this location's name in an
-    #   activity description
-    def regex_for_name
-      Friends::RegexBuilder.regex(@name)
+    # @return [String] a string representing the location's name and aliases
+    def to_s
+      unless @nicknames.empty?
+        nickname_str = " (" +
+                       @nicknames.map do |nickname|
+                         "#{NICKNAME_REFIX}#{Paint[nickname, :bold, :yellow]}"
+                       end.join(" ") + ")"
+      end
+
+      "#{Paint[@name, :bold]}#{nickname_str}"
+    end
+
+    # Add a alias, ignoring duplicates.
+    # @param nickname [String] the alias to add
+    def add_alias(nickname)
+      @nicknames << nickname
+      @nicknames.uniq!
+    end
+
+    # @param nickname [String] the alias to remove
+    # @raise [FriendsError] if the location does not have the given alias
+    def remove_alias(nickname)
+      unless @nicknames.include? nickname
+        raise FriendsError, "Alias \"#{nickname}\" not found for \"#{name}\""
+      end
+
+      @nicknames.delete(nickname)
+    end
+
+    # @return [Array] a list of all regexes to match the name in a string
+    # NOTE: Only full names and aliases
+    def regexes_for_name
+      # We generously allow any amount of whitespace between parts of a name.
+      splitter = "\\s+"
+
+      # Create the list of regexes and return it.
+      chunks = name.split(Regexp.new(splitter))
```

Comment:
I think this allow-any-amount-of-whitespace thing is fine but not actually necessary. (And honestly it was probably unnecessary for friend regexes too.) I'm wondering if something like:

```ruby
[name, @nicknames].map { |str| Friends::RegexBuilder.regex(str) }
```

is really what we want and simplifies a lot?

Reply:
Yeah the whitespace thing is not really needed. 

It should do the same thing and then the comment match more the line. I will use it,